app/models/concerns/action_plan/statistics.rb

```diff
@@ -0,0 +1,71 @@
+class ActionPlan < ActiveRecord::Base
+  module Statistics
```

Comment:
I would better extract this into a `ActionPlanStatisticsCaltulator` service that receives an `ActionPlan` and stores its `statistics`. That service could be used by the `Worker` to re-calculate statistics without polluting the model. I think they're two separate concerns.